File: 724FindPivotIndex.py
class PivotIndex:
    def pivotIndex(self, nums):
        summ = sum(nums)
        leftsum = 0
        for i, x in enumerate(nums):
            if leftsum == summ - leftsum - x:
                return i
            leftsum += x
        return -1

if __name__ == "__main__":
    solution = PivotIndex()
    print(solution.pivotIndex([1,7,3,6,5,6]))

File: 724FindPivotIndex.java
class PivotIndex {
    public int pivotIndex(int[] nums) {
        int sum = 0, leftsum = 0;
        for (int x : nums) {
            sum += x;
        }
        for (int i = 0; i < nums.length; i++) {
            
            if (leftsum == sum - leftsum - nums[i]) {
                return i;
            }
            leftsum += nums[i];
        }
        return -1;
    }

    public static void main(String[] args) {
        PivotIndex pivotIndex = new PivotIndex();
        int[] a = new int[] {1,7,3,6,5,6};
        System.out.println(pivotIndex.pivotIndex(a));
    }
}
